appscale/tools/node_layout.py

```diff
@@ -737,7 +740,20 @@ def from_locations_json_list(self, locations_nodes_list):
             # Locations JSON is incorrect if we get here.
             return None
           break
-
+    if len(nodes_copy) > 0:
+      for open_node in open_nodes:
+        for _, node in enumerate(nodes_copy):
+          # Match nodes based on jobs/roles
+          nodes_copy.remove(node)
+          roles = node.roles
+          node.from_json(open_node)
+          node.roles = roles
+          if node.is_valid():
+            nodes.append(node)
+          else:
+            # Locations JSON is incorrect if we get here.
+            return None
```

Comment:
@tmarballi we don't handle it because NodeLayout isn't really equipped to in my opinion, returning None makes the caller raise an exception (because exceptions are never raised in this file for some reason)